### PR TITLE
Add REST API parameter type documentation

### PR DESCRIPTION
**Pull request by jfim**
Add support to generate Swagger parameter type documentation and JSON
schemas, as well as support for examples. Add documentation for the
instance creation API.

### CODE REVIEW

File: pinot-common/pom.xml [hunk 1]
```diff
@@ -190,6 +190,10 @@
       <groupId>org.restlet.jse</groupId>
       <artifactId>org.restlet</artifactId>
     </dependency>
+    <dependency>
```
Comment (mayankshriv) [inline pinot-common/pom.xml]: I also see imports of 'org.codehaus.jackson' in various other parts of the code. We should pick one and use that uniformly.

File: pinot-common/src/main/java/com/linkedin/pinot/common/restlet/swagger/SwaggerResource.java [hunk 2]
```diff
@@ -170,7 +174,12 @@ private void generateSwaggerForFinder(JSONObject pathObject, String routePath, F
...
         ArrayList<JSONObject> parameters = new ArrayList<JSONObject>();
 
-        for (Annotation[] annotations : method.getParameterAnnotations()) {
+        Annotation[][] parameterAnnotations = method.getParameterAnnotations();
+        Class<?>[] parameterTypes = method.getParameterTypes();
+        for (int i = 0; i < parameterTypes.length; i++) {
+          Class<?> parameterType = parameterTypes[i];
+          Annotation[] annotations = parameterAnnotations[i];
+
           if (annotations.length != 0) {
```
Comment (mayankshriv) [inline pinot-common/src/main/java/com/linkedin/pinot/common/restlet/swagger/SwaggerResource.java]: Need 'else' part for this 'if'.

File: pinot-common/src/main/java/com/linkedin/pinot/common/restlet/swagger/SwaggerResource.java [hunk 3]
```diff
@@ -197,6 +227,93 @@ private void generateSwaggerForFinder(JSONObject pathObject, String routePath, F
     }
   }
 
+  private JSONObject schemaForType(Class<?> type) {
+    try {
+      JSONObject schema = new JSONObject();
+
+      schema.put("type", "object");
```
Comment (mayankshriv) [inline pinot-common/src/main/java/com/linkedin/pinot/common/restlet/swagger/SwaggerResource.java]: Use static final instead of hard-coded values

File: pinot-common/src/main/java/com/linkedin/pinot/common/restlet/swagger/SwaggerResource.java (hunk 2 above)
Comment (jfim) [inline pinot-common/src/main/java/com/linkedin/pinot/common/restlet/swagger/SwaggerResource.java]: This piece of code generates documentation based on the annotations (see the annotated Instance class). If there are no annotations on the parameter, then there isn't a place to pull the documentation from, and as such, we don't generate an empty documentation object in the documentation (which would probably not validate against the Swagger spec).

File: pinot-common/pom.xml (hunk 1 above)
Comment (jfim) [inline pinot-common/pom.xml]: Good idea, although this is out of scope for this PR. I have created PINOT-3192 to track this.

File: pinot-common/src/main/java/com/linkedin/pinot/common/restlet/swagger/SwaggerResource.java (hunk 3 above)
Comment (jfim) [inline pinot-common/src/main/java/com/linkedin/pinot/common/restlet/swagger/SwaggerResource.java]: What do you think would be a good name for this constant? All of those strings are from the Swagger spec and not reused anywhere else in the Pinot codebase.

**Comment by mcvsubbu** [comment]
lgtm
